tests/suites/main_test.function

```diff
@@ -33,6 +33,13 @@
 #include "psa/crypto.h"
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
+/* If MBEDTLS_PSA_CRYPTO_C is defined, make sure MBEDTLS_PSA_CRYPTO_CLIENT
+ * is defined to run tests with dependency on MBEDTLS_PSA_CRYPTO_CLIENT.
+ */
+#if defined(MBEDTLS_PSA_CRYPTO_C)
+#define MBEDTLS_PSA_CRYPTO_CLIENT
```

Comment:
Why isn't this somewhere in `include/*/*`?

Reply:
I put it there because it is only needed in the tests as it is not possible to declare a test dependency like "MBEDTLS_PSA_CRYPTO_C || MBEDTLS_PSA_CRYPTO_CLIENT". Otherwise I thought to put it in check_config.h and if we go that way I was thinking in having just "#if defined(MBEDTLS_PSA_CRYPTO_CLIENT)" to guard the code in psa_crypto_client.c instead of #if defined(MBEDTLS_PSA_CRYPTO_CLIENT) || defined(MBEDTLS_PSA_CRYPTO_C). Would you prefer it that way?

Reply:
I actually expected
```
#if defined(MBEDTLS_PSA_CRYPTO_C)
#define MBEDTLS_PSA_CRYPTO_CLIENT
#endif
```
in `include/psa/crypto_something.h`, or maybe in `check_config.h` but that doesn't work for applications that have their own `config.h`. And then both library code and tests would only check for `MBEDTLS_PSA_CRYPTO_CLIENT`, without having to care why the option was enabled.

Reply:
Moved to crypto_types.h.